subsys/bluetooth/audio/has.c

```diff
@@ -1122,12 +1132,23 @@ int bt_has_preset_name_change(uint8_t index, const char *name)
 }
 #endif /* CONFIG_BT_HAS_PRESET_SUPPORT */
 
-static int has_init(const struct device *dev)
+int bt_has_register(const struct bt_has_register_param *param)
 {
-	ARG_UNUSED(dev);
+	LOG_DBG("Trying to start HAS service...");
+	int err;
+
+	CHECKIF(!param) {
+		LOG_DBG("NULL params pointer");
+		return -EINVAL;
+	}
+
+	CHECKIF(param->hearing_aid_type > 2) {
```

Comment:
I'm not a fan of this magic number :)
I would assume that BT_HAS_HEARING_AID_TYPE_BANDED is the max of enum, and its very unlikely to change. Check this one instead, or create a BT_HAS_HEARING_AID_TYPE_INVALID as max and check?

Reply:
or just check all variants, as there are only 3 possibilities.

Reply:
> If `hearing_aid_type` is a an enum, we could/should consider not doing this check, right?

true - fixing